Sources/AsyncQueue/CancellableQueue.swift

```diff
@@ -0,0 +1,448 @@
+// MIT License
+//
+// Copyright (c) 2025 Dan Federman
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+
+/// A queue that is capable of cancelling incomplete tasks.
+public final class CancellableQueue<UnderlyingQueue: Sendable>: Sendable {
+	// MARK: Initialization
+
+	/// Instantiates a cancellable queue.
+	/// - Parameter underlyingQueue: The queue whose work can be cancelled.
+	public init(underlyingQueue: UnderlyingQueue) {
+		self.underlyingQueue = underlyingQueue
+	}
+
+	// MARK: Public
+
+	/// Cancels the currently executing task, as well as any task currently pending in the queue.
+	public func cancelTasks() {
```

Comment:
`cancelAll` is a reasonable name too. I like `cancelTasks` a little bit better because it doesn't imply that future tasks will be cancelled as well.